integrations/kube-agent-updater/pkg/controller/utils.go

```diff
@@ -19,11 +19,35 @@ package controller
 import (
 	"strings"
 
+	"github.com/docker/distribution/reference"
 	"github.com/gravitational/trace"
 	"golang.org/x/mod/semver"
 	v1 "k8s.io/api/core/v1"
 )
 
+func getWorkloadVersion(podSpec v1.PodSpec) (string, error) {
+	var current string
+	image, err := getContainerImageFromPodSpec(podSpec, teleportContainerName)
+	if err != nil {
+		return current, trace.Wrap(err)
+	}
+
+	imageRef, err := reference.ParseNamed(image)
+	if err != nil {
+		return current, trace.Wrap(err)
+	}
+	taggedImageRef, ok := imageRef.(reference.Tagged)
+	if !ok {
+		return "", trace.BadParameter("imageRef %s is not tagged", imageRef)
```

Comment:
Should we also check if the user is running a custom image that includes a temporary fix because it the official release was not ready/published?

Reply:
I don't think we want to block update based on the image. We discussed adding an annotation on the deployment/statefulset not to be reconciled but I forgot to add it. I'll do it when piping very trigger, version getter and image validator together.

Reply:
Got it. Looks better solution anw if you are forced to run a custom image